calico/network-policy/policy-rules/log-rules.mdx

```diff
@@ -0,0 +1,191 @@
+
+Logs can be found in different locations, depending on your system.
+You can often find Calico policy logs by:
+* using the `journalctl` utility.
+* searching in `/var/log/syslog`.
+* searching in `/var/log/kern.log`.
+
+Example of iptables logs:
+
+```bash
+2024-11-19T12:15:03.023805-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=eth0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=69.147.80.15 LEN=60 TOS=0x00 PREC=0x00 TTL=63 ID=58406 DF PROTO=TCP SPT=45588 DPT=80 WINDOW=64860 RES=0x00 SYN URGP=0
+
+2024-11-19T12:15:28.648818-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=eth0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=8.8.8.8 LEN=84 TOS=0x00 PREC=0x00 TTL=63 ID=48210 DF PROTO=ICMP TYPE=8 CODE=0 ID=1 SEQ=1
+
+2024-11-19T12:15:02.990083-08:00 c1-control kernel: calico-packet: IN=cali527b0801ecb OUT=calid3446e883f0 MAC=ee:ee:ee:ee:ee:ee:1e:0a:36:33:f5:09:08:00 SRC=172.16.193.134 DST=172.16.193.133 LEN=96 TOS=0x00 PREC=0x00 TTL=63 ID=19241 DF PROTO=UDP SPT=42896 DPT=53 LEN=76
+```
+
+#### Log output for eBPF
+
+BPF dataplane logs the final verdict applied to the packet once it matched ANY log rule.
+This means that the log entry reflects the ultimate decision (whether the traffic is allowed or denied) after evaluating the full set of applicable rules and policies.
+
+eBPF policy logs are sent to the trace pipe and can be viewed by using the following command:
+
+```bash
+kubectl exec -n calico-system -it ds/calico-node -- bpftool prog tracelog
+```
+
+The precise format of the eBPF trace logs depends on your kernel and sysctl settings.
+However, a typical log format includes information about the active process, a timestamp and then the IP header.
+
+Example of eBPF logs:
+
+```bash
+curl-5288    [000] ..s2.  3055.982021: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 6 src 172.16.193.131:56042 dest 69.147.80.12:80
+
+<...>-4950    [001] ..s2.  2960.748464: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 1 src 172.16.193.131:0 dest 8.8.8.8:8
+
+<...>-5288    [000] ..s2.  3055.954466: bpf_trace_printk: cali527b0801ecb-E: policy ALLOWED proto 17 src 172.16.193.131:36325 dest 10.43.0.10:53
+```
+
+$[prodname] eBPF logs use a custom format to output the information, here is the breakdown of this format:
+
+![curl](/img/calico/curl-bpf-log.svg)
+
+1. Process name and ID.
+   Be aware that the process name might not directly correspond to the responsible process for the traffic.
+2. CPU number, Kernel flags, timestamp, BPF print function used to log this message.
+3. Interface name.
+4. Traffic type.
+   In this example, egress (`E`= egress, `X`= XDP, `I`= ingress).
+5. Policy verdict (`ALLOWED`, `DENIED`).
+6. IP protocol number.
+   In this example, 6 is for TCP (1 ICMP, 17 UDP).
+7. Source IP and source port.
+8. Destination IP and destination port.
+
+## Prerequisites
```

Comment:
FYI, there's no mention of OSS 3.29 because we'll only be including this in the docs for 3.29+.